packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts

```diff
@@ -323,9 +324,31 @@ export class ModelTransformer extends TransformerModelBase implements Transforme
   };
 
   generateResolvers = (context: TransformerContextProvider): void => {
+    const dataSourceMapping = {};
     this.resourceGeneratorMap.forEach((generator) => {
       generator.generateResources(context);
+      Object.assign(
+        dataSourceMapping,
+        Object.fromEntries(
+          Object.entries(generator.getDatasourceMap())
+            .filter(([, datasource]) => datasource.ds.type === 'AMAZON_DYNAMODB')
+            .map(([modelName, datasource]) => {
+              if (datasource.ds.dynamoDbConfig && !cdk.isResolvableObject(datasource.ds.dynamoDbConfig)) {
```

Comment:
I'm not sure there is a case when `isResolvableObject` would be true.

Reply:
Why is the check necessary? My read of the this (which might be wrong--see comment above) is that we're trying to get to the table name, which might not be resolved at synth time? What happens if the DS is a resolvable--will we skip resolver generation? How would a customer resolve in that situation?

Reply:
The check is necessary for TypeScript here. `datasource.ds.dynamoDbConfig` is of type `DynamoDBConfigProperty | IResolvable`. 

Since we are creating the this with `AppSync.DataSource` ourselves we should know that it will always be `DynamoDBConfigProperty` and not `IResolvable`.

A more fault tolerant option would be to just print a warning for any table names that could not be resolved for whatever reason. The customer would then add this information to the mapping themselves in their Gen 2 project.

Reply:
It sounds like a "this shouldn't happen" check to satisfy the compiler then--thanks for the explanation. Printing warnings is probably the wrong way to go here, since we already have so many and we don't actually expect this to occur anyway, so let's keep it as an error.